libs/api/domains/license-service/src/lib/client/driving-license-client/drivingLicenseMappers.ts

```diff
@@ -0,0 +1,154 @@
+    case 'be':
+      return 'Bifreið með eftirvagn/tengitæki'
+    case 'c1e':
+      return 'C1E'
+    case 'd1e':
+      return 'D1E'
+    case 'ce':
+      return 'CE'
+    case 'de':
+      return 'DE'
+    default:
+      return licenseClass
+  }
+}
+
+export const drivingLicensesToSingleGenericLicense = (
+  licenses: GenericDrivingLicenseResponse[],
+): GenericUserLicense | null => {
+  if (licenses.length === 0) {
+    return null
+  }
+
+  // TODO(osk) we're only handling the first driving license, we get them ordered so pick first
+  const license = licenses[0]
+
+  // We parse license data into the fields as they're displayed on the physical drivers license
+  // see: https://www.samgongustofa.is/umferd/nam-og-rettindi/skirteini-og-rettindi/okurettindi-og-skirteini/
+  const data: Array<GenericLicenseDataField> = [
+    // We don't get the name split into two from the API, combine
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: '2. Eiginnafn 1. Kenninafn',
+      value: license.nafn,
+    } as GenericLicenseDataField,
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: '3. Fæðingardagur og fæðingarstaður',
+      // TODO(osk) parse nationalId into date of birth. helper util exists somewhere?
+      value: [license.kennitala ?? null, license.faedingarStadurHeiti ?? null]
+        .filter(Boolean)
+        .join(' '),
+    } as GenericLicenseDataField,
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: '4a. Útgáfudagur',
+      value: license.utgafuDagsetning,
+    } as GenericLicenseDataField,
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: '4b. Lokadagur',
+      value: license.gildirTil,
+    } as GenericLicenseDataField,
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: '4c. Nafn útgefanda',
+      value: license.nafnUtgafustadur,
+    } as GenericLicenseDataField,
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: '4d. Kennitala',
+      value: license.kennitala,
+    } as GenericLicenseDataField,
+    {
+      type: GenericLicenseDataFieldType.Value,
+      label: '5. Númer',
+      value: (license?.id ?? '').toString(),
+    } as GenericLicenseDataField,
+    {
+      type: GenericLicenseDataFieldType.Group,
+      label: '9. Réttindaflokkar',
+      fields: (license.rettindi ?? []).map((field) => ({
+        type: GenericLicenseDataFieldType.Category,
+        name: (field.nr ?? '').trim(),
+        label: licenseCategoryClassToLabel((field.nr ?? '').trim()),
+        fields: [
+          {
+            type: GenericLicenseDataFieldType.Value,
+            label: 'Lokadagur',
+            value: field.gildirTil,
+          },
+          {
+            type: GenericLicenseDataFieldType.Value,
+            label: 'Útgáfudagur',
+            value: field.utgafuDags,
+          },
+        ],
+      })),
+    } as GenericLicenseDataField,
+  ].filter((Boolean as unknown) as ExcludesFalse)
```

Comment:
Is this a logic error? Doesn't look like this array will ever contain false values.

Reply:
yes, this and the casts are redundant as is